src/software/ai/hl/stp/play/enemy_ball_placement/enemy_ball_placement_play_fsm.cpp

```diff
@@ -57,39 +57,56 @@ void EnemyBallPlacementPlayFSM::avoid(const Update& event)
     {
         if (contains(stadium, robot.position()))
         {
-            Vector lateral_positioning_vector =
+            // The "backbone" of the stadium connects the ball position and the placement
+            // point. A robot located in the stadium will move perpendicular to the
+            // backbone to be outside the stadium. There are two possible destination
+            // points, one on each side of the stadium
+
+            // This unit vector is perpendicular to the backbone, and determines the
+            // direction of robot motion
+            Vector lateral_positioning_unit_vector =
                 placement_point_to_ball.perpendicular().normalize();
+            // This vector represents the robot's longitudinal position along the
+            // backbone, relative to the placement point
             Vector longitudinal_positioning_vector =
                 (robot.position() - placement_point).project(placement_point_to_ball);
             Point p1 = placement_point + longitudinal_positioning_vector +
-                       lateral_positioning_vector * distance_to_keep;
+                       lateral_positioning_unit_vector * distance_to_keep_meters;
             Point p2 = placement_point + longitudinal_positioning_vector -
-                       lateral_positioning_vector * distance_to_keep;
+                       lateral_positioning_unit_vector * distance_to_keep_meters;
             Point destination;
             Rectangle fieldLines = world_ptr->field().fieldLines();
-            if (!contains(fieldLines, p1))
+            // If either destination point is outside the field, then pick the other
+            // point. It is impossible for both points to be outside the field
```

Comment:
Isn't it technically possible for both p1 and p2 (shown in red) to be outside of the field if the ball is in the corner and the placement position is somewhere towards the center? (the straight line shows the "backbone")
![image](https://github.com/UBC-Thunderbots/Software/assets/28585597/14a4ec81-f26d-4e68-9172-76db56b8427f)

Though in either case we can _probably_ ignore this edge case